### PR TITLE
tests: kernel: Add description for tests in common, interrupt and boot_page_table

### PR DESCRIPTION
**Pull request by spoorthik**
Add description and doxygen links for kernel test cases

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

### CODE REVIEW

**Comment by nashif** [comment]
looks good, but we need to add doxygen groups to contain all of those tests.

File: tests/kernel/arm_irq_vector_table/src/arm_irq_vector_table.c [hunk 1]
```diff
@@ -64,14 +55,23 @@ void isr2(void)
 }
 
 /**
- * @brief test installation of ISRs directly in the vector table
+ * @defgroup kernel_interrupt Interrupt_tests
```
Comment (nashif) [inline tests/kernel/arm_irq_vector_table/src/arm_irq_vector_table.c]: should add a suffix (_tests) to the group to differentiate this from the actual feature documentation, so this becomes kernel_interrupt_tests

Comment (nashif) [reply tests/kernel/arm_irq_vector_table/src/arm_irq_vector_table.c]: should be 

`@defgroup kernel_interrupt Interrupts`

first argument is the identifier, the second is the title in the documentation

File: tests/kernel/boot_page_table/src/boot_page_table.c [hunk 2]
```diff
@@ -82,7 +82,22 @@ static void ending_start_addr_range(u32_t start_addr_range)
 			      addr_range, REGION_PERM);
 	}
 }
+/**
+ * @brief Test for boot page table
+ * @defgroup kernel_boot_page_table boot_page_table_tests
```
Comment (nashif) [inline tests/kernel/boot_page_table/src/boot_page_table.c]: `@defgroup kernel_boot_page_table_tests Boot Page Table`

We probably want to put this in another category with other tests.

File: tests/kernel/common/src/main.c [hunk 3]
```diff
@@ -36,7 +36,18 @@ void test_printk(void)
 }
 #endif
 
+/**
+ * @defgroup kernel_common common_tests
```
Comment (nashif) [inline tests/kernel/common/src/main.c]: @defgroup kernel_common_tests Common Tests

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by nashif** [comment]
Also see https://github.com/zephyrproject-rtos/zephyr/wiki/Documenting-Code-for-Traceability

**Comment by spoorthik** [comment]
@nashif : Addressed review comments. Please have a look.

File: tests/kernel/arm_runtime_nmi/src/arm_runtime_nmi.c [hunk 4]
```diff
@@ -30,6 +26,14 @@ static void nmi_test_isr(void)
 	TC_END_REPORT(TC_PASS);
 }
 
+/**
+ * @brief Test the behavior of CONFIG_RUNTIME_NMI at runtime.
+ * @addtogroup kernel_interrupt
```
Comment (nashif) [inline tests/kernel/arm_runtime_nmi/src/arm_runtime_nmi.c]: kernel_interrupt_tests

File: tests/kernel/common/src/atomic.c [hunk 5]
```diff
@@ -7,6 +7,20 @@
 #include <ztest.h>
 #include <atomic.h>
 
+/**
+ * @addtogroup kernel_common
```
Comment (nashif) [inline tests/kernel/common/src/atomic.c]: kernel_common_tests

File: tests/kernel/common/src/bitfield.c [hunk 6]
```diff
@@ -14,6 +14,19 @@
 #define BIT_VAL(bit)    (1 << (bit & 0x7))
 #define BITFIELD_SIZE   512
 
+/**
+ * @addtogroup kernel_common
```
Comment (nashif) [inline tests/kernel/common/src/bitfield.c]: kernel_common_tests, and everything below.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/common/src/main.c [hunk 7]
```diff
@@ -20,6 +20,14 @@ extern void test_timeout_order(void);
 extern void test_clock_cycle(void);
 extern void test_clock_uptime(void);
 extern void test_multilib(void);
+
+/**
+ * @defgroup kernel_common_tests common Tests
```
Comment (nashif) [inline tests/kernel/common/src/main.c]: s/common Tests/Common Tests/

File: tests/kernel/common/src/main.c [hunk 8]
```diff
@@ -20,6 +20,14 @@ extern void test_timeout_order(void);
...
+ * @ingroup all_tests
+ * @{
```
Comment (nashif) [inline tests/kernel/common/src/main.c]: when you do this, everything is added to doxygen, also non-tests, either move the helper functions out of the brackets or use `@ingroup kernel_common_tests` individually on each documented test. Here for example delayed_resubmit_work_handler  needs to be moved out or excluded.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/common/src/main.c (hunk 8 above)
Comment (spoorthik) [reply tests/kernel/common/src/main.c]: @nashif : I think there is no such helper functions to be moved out. All are test functions.

**Comment by spoorthik** [comment]
@nashif: Can you please review latest changes?

**Review by nashif — APPROVED** [review]

**Pull request merged**